lightning/src/ln/channel.rs

```diff
@@ -1716,8 +1764,14 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 				removed_outbound_total_msat += htlc.amount_msat;
 			}
 		}
-		if htlc_inbound_value_msat + msg.amount_msat + self.value_to_self_msat > (self.channel_value_satoshis - Channel::<ChanSigner>::get_our_channel_reserve_satoshis(self.channel_value_satoshis)) * 1000 + removed_outbound_total_msat {
-			return Err(ChannelError::Close("Remote HTLC add would put them over their reserve value"));
+
+		// The + 1 is for the HTLC that is currently being added to the commitment tx.
+		let remote_fee_cost_msat = if self.channel_outbound { 0 } else {
+			self.commit_tx_fee_msat(self.htlc_count_next_remote_commit_tx() + 1)
+		};
```

Comment:
I think the complexity of this can be hidden in a method that:
- is used in replace of `htlc_count_next_remote_commit_tx` to calculate `remote_fee_cost_msat`,
- possibly accepts a parameter for additional HTLCs,
- checks `channel_outbound`, and
- calls `commit_tx_fee_msat`

Essentially, move everything inside `htlc_count_next_remote_commit_tx` and rename it.

Likewise for `htlc_count_next_local_commit_tx`.

Reply:
Gotcha, I had something like that previously but @TheBlueMatt preferred doing the math within `update_add`/`send_htlc`: https://github.com/rust-bitcoin/rust-lightning/pull/577#discussion_r407825688 because the old function had an off-by-one that was non-obvious.

Although, I think the additional parameter for additional HTLCs could solve that off-by-one problem. & I'm generally in favor of farming more logic out to helpers. @TheBlueMatt does this sound good?

Reply:
Yea, I'm fine with it. As long as the end state is super readable and well-documented.

Reply:
Lmk the helpers stll need work!